org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/VariableDebugAdapter.java

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter;
+
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.ls.debug.internal.Logger;
+
+import com.sun.jdi.AbsentInformationException;
+import com.sun.jdi.IncompatibleThreadStateException;
+import com.sun.jdi.Location;
+import com.sun.jdi.Method;
+import com.sun.jdi.StackFrame;
+import com.sun.jdi.ThreadReference;
+
+public class VariableDebugAdapter {
+
+    private final DebugAdapter parent;
+    private final IdCollection<StackFrame> frameCollection = new IdCollection<>();
+
+    public VariableDebugAdapter(DebugAdapter debugAdapter) {
+        this.parent = debugAdapter;
```

Comment:
>parent [](start = 13, length = 6)

Question, what's the relationship between VariableDebugAdapter and DebugAdapter? the naming makes me think about inheritance and actually it isn't